copier/tools.py

```diff
@@ -172,3 +173,56 @@ def cleanup(self):
     @staticmethod
     def _robust_cleanup(name):
         shutil.rmtree(name, ignore_errors=False, onerror=handle_remove_readonly)
+
+
+def get_lines(content: bytes) -> StrSeq:
+    # Assume UTF-8 encoding for text files. Since Jinja2 is already doing this
+    # for templates, there  probably isn't much harm in also doing it for all
+    # text files. The worst case should be a failure to show the diffs. An
+    # alternative is to add chardet support.
+    return content.decode("utf-8").split(os.linesep)
+
+
+def is_binary(content: bytes) -> bool:
+    """Return a boolean indicating if the content is a binary file."""
+    return b"\x00" in content or b"\xff" in content
+
+
+def color_print(color: int, msg: str):
+    print(f"{color}{msg}{colorama.Fore.RESET}")
+
+
+def print_diff(dst_relpath: Path, original_content: bytes, new_content: bytes):
+    try:
+        original_lines = get_lines(original_content)
+    except Exception:
+        color_print(
+            colorama.Fore.YELLOW, "diff unavailable: unable to determine encoding"
+        )
+        return
+    try:
+        new_lines = get_lines(new_content)
+    except Exception:
+        color_print(
+            colorama.Fore.YELLOW, "diff unavailable: unable to determine encoding"
+        )
+        return
```

Comment:
I know some flake8 plugins would say: "only one line in a try block", but this could probably be factorized to this:

```suggestion
    try:
        original_lines = get_lines(original_content)
        new_lines = get_lines(new_content)
    except Exception:
        color_print(
            colorama.Fore.YELLOW, "diff unavailable: unable to determine encoding"
        )
        return
```

Reply:
Also, is there a reason you catch `Exception` rather than the more specific `UnicodeDecodeError`?